src/utils/apiFetch.ts

```diff
@@ -0,0 +1,26 @@
+const getCsrfToken = (): string => {
+  const token = document.cookie
+    .split('; ')
+    .find((row) => row.startsWith('XSRF-TOKEN='))
+    ?.split('=')[1];
+  return token || '';
+};
+
+const apiFetch = async (
+  url: string,
+  options?: RequestInit
+): Promise<Response> => {
+  const csrfToken = getCsrfToken();
+
+  const updatedOptions: RequestInit = {
+    ...options,
+    headers: {
+      ...options?.headers,
+      'X-XSRF-TOKEN': csrfToken,
```

Comment:
@M0NsTeRRR should I rename this to `XSRF-TOKEN`?

Reply:
Yes as stated by [rfc6648](https://datatracker.ietf.org/doc/html/rfc6648) and csurf supports it https://www.npmjs.com/package/csurf#value